### ISSUE TITLE
Stop rendering pseudo-italic fonts when italic is not available

### ISSUE THREAD

**Issue by ivancuric** [Microsoft/vscode#25895]
When using fonts which don't have italic variants, such as Fira Mono, VScode renders them anyway, using a simple skew.

This looks quite bad, and should be avoided. 

- VSCode Version: 1.11.2
- OS Version: MacOS Sierra

**Comment by DevotionGeo** [comment]
@ivancuric While not the solution you wanted, but DejaVu Sans Mono looks very similar to Fira Mono with nice italic variants.

**Comment by t-ricci-enhancers** [comment]
Yeah, emojis and symbols are really bad with italics (I have a theme that makes all comments italics).
Can this be fixed please?

**Comment by DevotionGeo** [comment]
If from "Can this be fixed" you meant to unitalicize the comments but keep using that theme, add the following to your settings.json

```
"editor.tokenColorCustomizations": {
    "[Theme Name]": {
      "textMateRules": [
        {
          "scope": [
            "comment",
            "comment.block",
            "comment.block.documentation",
            "comment.line",
          ],
          "settings": {
            "fontStyle": ""
          },
        }
      ]
    }
}
```

**Comment by ivancuric** [comment]
Nothing to do with a specific token, but font rendering in general.
If a certain glyph doesn't have an italic variant, VScode shouldn't try to render it using faux italic. Same goes for faux bold.

This might be a Chromium issue.

**Comment by babramczyk** [comment]
Yup -- emojis are particularly bad

![image](https://user-images.githubusercontent.com/16885191/98749930-5c86f080-2382-11eb-8995-8a2442a9743a.png)

**Comment by REYNEP** [comment]
Yes, Emoji's LOOK HORRIBLE.... 😢

**Comment by retr0c0de** [comment]
Is there any way to make emoji non-responsive to the font-style?

**Comment by REYNEP** [comment]
For now, I found a WorkAround if anyone is Interested,     USING TextMate-Grammer that VSCODE has Support for:
>Its Pretty much HardCoded based on What Characters are what..... Not really related to Fonts having the Italic Variations or not.... But at least it can FIX THE Emojis 😉

<details>
<summary>package.json    (in an extension)</summary>

```json
//package.json

"contributes": {
    "themes": [
        {
            "doc": "EXAMPLE THEME",
            "label": "Eva Dark",
            "uiTheme": "vs-dark",
            "path": "./themes/Eva-Dark.json"
        }
    ],
    "grammars": [
        {
            "doc": "as this is package.json   we have syntaxes folder which has another json",
            "path": "./syntaxes/test.tmGrammer.json",

            "doc": "should be matching to    'scopeName' that you have in test.tmGrammer.json",
            "scopeName": "NonASCII.injection.cpp",

            "doc":  "We want to INJECT, not OverWrite the Existing TEXT-MATE Scopes",
            "injectTo": ["source.cpp"]
        }
    ]
},
```
</details>

<details>
<summary>related TextMateGrammer.json</summary>

```json
// ./syntaxes/test.tmGrammer.json
// I know json doesn't support DOC-STRINGS

// Well, why not just highlight nonASCII chars       U can (non-)highlight your Main Language too if Unicode has that support
{
  "scopeName": "NonASCII.injection.cpp",
  "injectionSelector": [
    "L:comment.block.documentation.cpp",
    "L:comment.line.double-slash.cpp"
  ],
  "patterns": [
    {
      "include": "#Non-ASCII-Chars"
    }
  ],
  "repository": {
    "Non-ASCII-Chars": {
      "doc": "Note the double BACKSLASHES ;)"
      "match": "\\p{^ASCII}",
      "name": "comment.non-ascii.cpp"
   }
}
```
</details>

**Then for any theme that makes COMMENTS Italic     you can use the scope that you just created, in my case that would be**  `comment.non-ascii.cpp`

Well, you can choose any of the 'Character Property [3.]'    from     [TextMate Expressions](https://macromates.com/manual/en/regular_expressions)

Here is the Semantic Guide part where you should start: https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#injection-grammars

**Comment by robbiemu** [comment]
This may be an operating system issue?

I'm on Ventura 13.1 on a mac, and I wanted an italic from one font but the Bergen Mono otherwise. I installed only the Bergen Mono non-italics, but found it rendered with an artificial italic. Having found this thread, I gave up, and went to install the 3 missing italics. The Font installer warns me that a version of this font is already installed, and renders the same fake italics from the Bergen Mono that I believe I am seeing in vscode.

The proper solution is to remove the font, then reinstall with the additional missing faces (otherwise I believe if you try "keep both" it will rename them). But what surprised me was that it detected a conflict for a font based on only new face-and-variant of the font.

**Comment by ivancuric** [comment]
VScode should use `font-synthesis: none;` to fix this.

https://developer.mozilla.org/en-US/docs/Web/CSS/font-synthesis

**Comment by ZHDreamer** [comment]
Simply add the css option `font-synthesis: none;` can fix this issue:
![with_synthesis](https://github.com/microsoft/vscode/assets/40716043/21d8cd2f-02a6-4f8f-8c23-db1fe22d26bc)
![without_synthesis](https://github.com/microsoft/vscode/assets/40716043/db5f1084-4c7b-4be6-bb08-eecce0a10ab7)
I think add an option or enable us customize the css style of the font can easily solve this problem. Does any one working on this?

**Comment by Yuki-Codes** [comment]
> Simply adding the css option `font-synthesis: none;` can fix this issue

You can do this with the extension [vscode-custom-css](https://marketplace.visualstudio.com/items?itemName=be5invis.vscode-custom-css), but the entire extension feels like a bit of a hack.

![Image](https://github.com/user-attachments/assets/8a71e300-2a3e-4462-8ae9-07ee01eb5aeb)
